.github/workflows/test-e2e.yaml

```diff
@@ -0,0 +1,76 @@
+name: E2E Tests
+
+on:
+  workflow_call:
+
+jobs:
+  build-docker-image:
+    name: Build Docker image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build and export
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          tags: vechain/thor:${{ github.sha }}
+          outputs: type=docker,dest=/tmp/vechain-thor.tar
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: vechain-thor-image-${{ github.sha }}
+          path: /tmp/vechain-thor.tar
+          retention-days: 7
+
+  run-tests:
+    runs-on: ubuntu-latest
+    needs: build-docker-image
+    env:
+      THOR_IMAGE: vechain/thor:${{ github.sha }}
+    name: Run Tests
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          repository: vechain/thor-e2e-tests
+          # TODO: Should we use tags here? eg 'v2.1.0'
+          ref: main
```

Comment:
Perhaps need to think about this, wdyt @leszek-vechain

Reply:
I think `main` probably isn't the best, as we could create new tests that start breaking the thor CI.

IMO a commit SHA could be a good option for this. Once you write code / break existing tests, you should modify the e2e, create a PR, merge it, and then update the SHA in the GHA.

The same approach would have to be taken in the context of submodules too, since you would have to update e2e and then update the submodule reference.

So I could set the value to [fba3138](https://github.com/vechain/thor-e2e-tests/commit/fba3138cc140c8f734f68bdd4bbd6d27b670d7e5), which is the latest commit

Reply:
need to think about it - it seems like it will be quite a lot of work, maybe we can just move e2e to thor repo, in that case, you change code, you change tests in some commit